### PR TITLE
[#60] [Integrate] As a user, I can see the app cache and load survey list in Home screen

### PR DESCRIPTION
**Pull request by luongvo**
Resolves https://github.com/luongvo/flutter-survey/issues/60

## What happened 👀

- Setup local storage to load and save Survey list with [Hive](https://pub.dev/packages/hive).
- Setup `SurveyBoxHelper` to save survey list after fetching in `SurveyRepository`.
- Load cache survey list from Hive local storage when initializing `Home` screen.
 
## Insight 📝

N/A
 
## Proof Of Work 📹

- The app loads cache surveys with 3 items before fetching and showing the latest survey list from API.

https://user-images.githubusercontent.com/16315358/183493330-ee94711f-cee6-4106-8745-abeeb7766452.mp4

### CODE REVIEW

File: lib/usecase/base/base_use_case.dart [hunk 1]
```diff
@@ -17,3 +17,9 @@ abstract class NoParamsUseCase<T> extends BaseUseCase<Result<T>> {
 
   Future<Result<T>> call();
 }
+
+abstract class SimpleUseCase<T> {
```
Comment (markgravity) [inline lib/usecase/base/base_use_case.dart:21]: I think for better naming we should do some refactor here a bit like this

```dart
abstract class BaseUseCase<T extends Result> {
  const BaseUseCase();
}

abstract class FutureUseCase<T, P> extends BaseUseCase<Result<T>> {
  const UseCase() : super();

  Future<Result<T>> call(P params);
}

abstract class NoParamsFutureUseCase<T> extends BaseUseCase<Result<T>> {
  const NoParamsUseCase() : super();

  Future<Result<T>> call();
}

abstract class UseCase<T> {
  const UseCase();

  T call();
}
```

What do you think?

**Review by markgravity — COMMENTED** [review]
The rest LGTM 👍

File: lib/usecase/base/base_use_case.dart (hunk 1 above)
Comment (luongvo) [reply lib/usecase/base/base_use_case.dart:21]: @markgravity `UseCase` is most common used and the main usecase. Renaming it to `FutureUseCase` leads to a long base class name. I think I will keep the current one.

**Review by markgravity — APPROVED** [review]
LGTM 👍

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**